cmd/registry-bigquery/index/info.go

```diff
@@ -0,0 +1,168 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package index
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/apigee/registry-experimental/pkg/yamlquery"
+	"github.com/apigee/registry/pkg/connection"
+	"github.com/apigee/registry/pkg/mime"
+	"github.com/apigee/registry/pkg/names"
+	"github.com/apigee/registry/pkg/visitor"
+	"github.com/apigee/registry/rpc"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+func infoCommand() *cobra.Command {
+	var filter string
+	var project string
+	var dataset string
+	var step int
+	cmd := &cobra.Command{
+		Use:   "info PATTERN",
+		Short: "Build a BigQuery index of API information",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			c, err := connection.ActiveConfig()
+			if err != nil {
+				return err
+			}
+			pattern := c.FQName(args[0])
+			adminClient, err := connection.NewAdminClientWithSettings(ctx, c)
+			if err != nil {
+				return err
+			}
+			registryClient, err := connection.NewRegistryClientWithSettings(ctx, c)
+			if err != nil {
+				return err
+			}
+			if project == "" {
+				project = c.Project
+			}
+			client, err := bigquery.NewClient(ctx, project)
+			if err != nil {
+				return err
+			}
+			ds, err := getOrCreateDataset(ctx, client, dataset)
+			if err != nil {
+				return err
+			}
+			table, err := getOrCreateTable(ctx, ds, "info", info{})
+			if err != nil {
+				return err
+			}
+			v := &infoVisitor{
+				registryClient: registryClient,
+			}
+			err = visitor.Visit(ctx, v, visitor.VisitorOptions{
+				RegistryClient:  registryClient,
+				AdminClient:     adminClient,
+				Pattern:         pattern,
+				Filter:          filter,
+				ImplicitProject: &rpc.Project{Name: "projects/implicit"},
+			})
+			if err != nil {
+				return err
+			}
+			u := table.Inserter()
+			log.Printf("uploading %d info sets", len(v.infos))
+			for start := 0; start < len(v.infos); start += step {
+				log.Printf("%d", start)
+				end := min(start+step, len(v.infos))
+				if err := u.Put(ctx, v.infos[start:end]); err != nil {
+					return err
+				}
+			}
+			return nil
+		},
+	}
+	cmd.Flags().StringVar(&filter, "filter", "", "Filter selected resources")
+	cmd.Flags().StringVar(&project, "project", "", "Project to use for BigQuery upload (defaults to registry project)")
+	cmd.Flags().StringVar(&dataset, "dataset", "registry", "BigQuery dataset name")
+	cmd.Flags().IntVar(&step, "step", 10000, "Step size to use when uploading records to BigQuery")
```

Comment:
Would "batch size" be a more standard and universally understood term?

Reply:
Thanks, I was stuck trying to think of a single-word for this, do you think "batch" would work?

Reply:
Well, I don't think "--batch-size" is bad... but yes, I think "batch" would be clearer than "step".

Reply:
ok, `--batch-size` SGTM